scipy/special/tests/test_support_alternative_backends.py

```diff
@@ -47,20 +47,30 @@ def test_rel_entr_generic(dtype):
     xp_assert_close(res, xp.asarray(ref), xp=xp)
 
 
+class RNGKeeper:
```

Comment:
This addresses https://github.com/scipy/scipy/pull/20967#discussion_r1641914890. If you can think of another way using features of `pytest`, LMK.